cmd/juju/application/deploy_test.go

```diff
@@ -1034,140 +1059,16 @@ func (s *charmstoreSuite) runDeploy(c *gc.C, args ...string) error {
-	testcharms.UploadCharm(c, s.client, "trusty/mysql-0", "mysql")
-	testcharms.UploadCharm(c, s.client, "trusty/wordpress-1", "wordpress")
-
-	// Run the tests.
-	for i, test := range deployAuthorizationTests {
-		c.Logf("test %d: %s", i, test.about)
-
-		// Upload the charm or bundle under test.
-		url := charm.MustParseURL(test.uploadURL)
-		if url.Series == "bundle" {
-			url, _ = testcharms.UploadBundle(c, s.client, test.uploadURL, "wordpress-simple")
-		} else {
-			url, _ = testcharms.UploadCharm(c, s.client, test.uploadURL, "wordpress")
-		}
-
-		// Change the ACL of the uploaded entity if required in this case.
-		if test.readPermUser != "" {
-			s.changeReadPerm(c, url, test.readPermUser)
-		}
-		err := runDeploy(c, test.deployURL, fmt.Sprintf("wordpress%d", i))
-		if test.expectError != "" {
-			c.Check(err, gc.ErrorMatches, test.expectError)
-			continue
-		}
-		c.Assert(err, jc.ErrorIsNil)
-	}
-}
-
-func (s *DeployCharmStoreSuite) TestDeployWithTermsSuccess(c *gc.C) {
-	_, ch := testcharms.UploadCharm(c, s.client, "trusty/terms1-1", "terms1")
-	_, stdErr, err := runDeployWithOutput(c, "trusty/terms1")
-	c.Assert(err, jc.ErrorIsNil)
-	expectedOutput := `
-Located charm "cs:trusty/terms1-1".
-Deploying charm "cs:trusty/terms1-1".
-Deployment under prior agreement to terms: term1/1 term3/1
-`
-	c.Assert(stdErr, gc.Equals, strings.TrimSpace(expectedOutput))
-	s.assertCharmsUploaded(c, "cs:trusty/terms1-1")
-	s.assertApplicationsDeployed(c, map[string]applicationInfo{
-		"terms1": {charm: "cs:trusty/terms1-1", config: ch.Config().DefaultSettings()},
-	})
-	_, err = s.State.Unit("terms1/0")
-	c.Assert(err, jc.ErrorIsNil)
-}
-
-func (s *DeployCharmStoreSuite) TestDeployWithTermsNotSigned(c *gc.C) {
-	s.termsDischargerError = &httpbakery.Error{
-		Message: "term agreement required: term/1 term/2",
-		Code:    "term agreement required",
-	}
-	testcharms.UploadCharm(c, s.client, "quantal/terms1-1", "terms1")
-	err := runDeploy(c, "quantal/terms1")
-	expectedError := `Declined: some terms require agreement. Try: "juju agree term/1 term/2"`
-	c.Assert(err, gc.ErrorMatches, expectedError)
-}
-
-func (s *DeployCharmStoreSuite) TestDeployWithChannel(c *gc.C) {
-	ch := testcharms.Repo.CharmArchive(c.MkDir(), "wordpress")
-	id := charm.MustParseURL("cs:~client-username/precise/wordpress-0")
-	err := s.client.UploadCharmWithRevision(id, ch, -1)
-	c.Assert(err, gc.IsNil)
-
-	err = s.client.Publish(id, []csclientparams.Channel{csclientparams.EdgeChannel}, nil)
-	c.Assert(err, gc.IsNil)
-
-	err = runDeploy(c, "--channel", "edge", "~client-username/wordpress")
-	c.Assert(err, gc.IsNil)
-	s.assertCharmsUploaded(c, "cs:~client-username/precise/wordpress-0")
-	s.assertApplicationsDeployed(c, map[string]applicationInfo{
-		"wordpress": {charm: "cs:~client-username/precise/wordpress-0", config: ch.Config().DefaultSettings()},
-	})
-}
+// TODO
+//func (s *DeployCharmStoreSuite) TestDeployWithChannel(c *gc.C) {
```

Comment:
Let's look to re-implement this as discussed live

Reply:
100%